ambry-api/src/main/java/com/github/ambry/quota/EnforcementRecommendation.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.quota;
+
+/**
+ * Interface representing enforcement recommendation made by a {@link RequestQuotaEnforcer} or {@link HostQuotaEnforcer}
+ * implementation. QuotaEnforcer implementations can use this object to provide a boolean recommendation to throttle the
+ * request or not, along with usage information like usage percentage, name of the enforcer that made this recommendation,
+ * the recommended http status (indicating whether or not throttled request should be retried) and cost to serve the request.
+ */
+public interface EnforcementRecommendation {
```

Comment:
do we really need this one to be an interface? can we just use AmbryEnforcementRecommendation every where?

Reply:
I want to leave it as an interface for now because different implementations might want to implement it differently.